validation/views.py

```diff
@@ -707,6 +734,59 @@ def save_wrong_word(request, recording_id):
     return response
 
 
+@login_required()
+def record_audio(request):
+    if request.method == "POST":
+        form = RecordNewPhrase(request.POST, request.FILES)
+        translation = request.POST.get("translation")
+        transcription = request.POST.get("transcription")
+        audio_data = request.FILES["audio_data"]
+
+        phrase = Phrase.objects.filter(transcription=transcription).first()
+        if not phrase:
+            phrase = Phrase(
+                translation=translation,
+                field_transcription=transcription,
+                transcription=transcription,
+                kind=Phrase.SENTENCE if " " in transcription else Phrase.WORD,
+                status=Phrase.USER,
+                date=datetime.datetime.now(),
+                modifier=request.user,
+            )
+            phrase.save()
+
+        speaker = Speaker.objects.filter(user=request.user).first()
+        if not speaker:
+            speaker = Speaker(
+                full_name=request.user.first_name + " " + request.user.last_name,
+                code=request.user.first_name[:3].upper(),
+            )
+            speaker.save()
+
+        _id = create_new_rec_id(phrase, speaker)
+        audio_data.name = _id + ".wav"
+        rec = Recording(
+            id=_id,
+            compressed_audio=audio_data,
+            speaker=speaker,
+            phrase=phrase,
+            timestamp=0,
+            comment=f"Uploaded by user: {request.user}",
+            is_user_submitted=True,
+        )
+        rec.save()
+        source = settings.RECVAL_AUDIO_PREFIX + _id + ".wav"
+        dest = settings.RECVAL_AUDIO_PREFIX + _id + ".m4a"
```

Comment:
I find the variable name `_id` confusing, because `_something` is usually a semi-private instance variable, not a local variable. Presumably this is to avoid shadowing python’s built-in `id` method? What about `rec_id` instead?

```suggestion
        rec_id = create_new_rec_id(phrase, speaker)
        audio_data.name = rec_id + ".wav"
        rec = Recording(
            id=rec_id,
            compressed_audio=audio_data,
            speaker=speaker,
            phrase=phrase,
            timestamp=0,
            comment=f"Uploaded by user: {request.user}",
            is_user_submitted=True,
        )
        rec.save()
        source = settings.RECVAL_AUDIO_PREFIX + rec_id + ".wav"
        dest = settings.RECVAL_AUDIO_PREFIX + rec_id + ".m4a"
```

Reply:
Done, `rec_id` is much better.